src/Microsoft.ML.Data/Evaluators/EvaluatorBase.cs

```diff
@@ -510,7 +510,9 @@ protected PerInstanceEvaluatorBase(IHostEnvironment env, ModelLoadContext ctx,
                 throw Host.ExceptSchemaMismatch(nameof(schema), "score", ScoreCol);
         }
 
-        public virtual void Save(ModelSaveContext ctx)
+        void ICanSaveModel.Save(ModelSaveContext ctx) => SaveModel(ctx);
+
+        private protected virtual void SaveModel(ModelSaveContext ctx)
```

Comment:
>SaveModel [](start = 39, length = 9)

<summary>
Derived class, for example `A`, should overwrite 'SaveModel` so that `((ICanSaveModel)A).Save(ctx)` can correctly dump `A`.
</summary>

Reply:
Thank you !

---
In reply to: [254968381](https://github.com/dotnet/machinelearning/pull/2431#discussion_r254968381) [](ancestors = 254968381)